src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -385,76 +385,79 @@ def submit(self, jobs, info=None):
         # unordered list of jobs with random sandboxes.
         # We intend to sort them by sandbox.
 
-        # We also need to do a secondary sort by number of cores
-        # because we can have jobs with different number of cores
-        # and the same sandbox and they cannot be submitted together.
+        # Do a secondary sort for numberOfCores and also for
+        # highIO jobs, they need to be submitted separately.
 
         submitDict = {}
         nSubmits   = 0
         for job in jobs:
             sandbox = job['sandbox']
             numberOfCores = job.get('numberOfCores', 1)
+            highIOjob = job.get('highIOjob', False)
             if not sandbox in submitDict.keys():
                 submitDict[sandbox] = {}
             if not numberOfCores in submitDict[sandbox]:
-                submitDict[sandbox][numberOfCores] = []
-            submitDict[sandbox][numberOfCores].append(job)
+                submitDict[sandbox][numberOfCores] = {}
+            if not highIOjob in submitDict[sandbox][numberOfCores]:
+                submitDict[sandbox][numberOfCores][highIOjob] = []
```

Comment:
Also, the balance from non highIO to highIO jobs should be (much) more than 10 to 1, so again this means the performance impact should be small.